internal/terraform/node_resource_plan_instance.go

```diff
@@ -157,12 +157,32 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 
 	importing := n.importTarget.ID != ""
 
+	if importing && n.Config == nil && !n.generateConfig {
+		// Then the user wrote an import target to a target that didn't exist.
+		if n.Addr.Module.IsRoot() {
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Import block target does not exist",
```

Comment:
This seems like something which should be handled a lot earlier in validation, rather than plumbing the option all the way down to the resource node.

Reply:
Yes, I agree.

We have a separate ticket for handling the import validation, which I think CJ is working on. Ideally, once that is done the statements will never be triggered but I didn't want to push code that panics on the off chance the validation work doesn't get finished in time for the beta.

In the end the code in [transform_config](https://github.com/hashicorp/terraform/pull/33186/files#diff-eae4ab33ad5bab5fbde2fcc9be6715f9fb067a051d73d7256f520825a4f20daaR159-R184) would just return a basic error that says "there's a problem, it should have been caught earlier. this is a bug in terraform; please report it" and we could remove the diagnostics from here.

I didn't want to implement that now because it gives a really bad error message (ie. no config context, no diagnostic printing) and put these in as a backstop while we wait for proper validation.

I did have a look at implementing something similar to the moved blocks validation in the refactoring package but I found it was very in depth (for example you need to evaluate for each and count expressions for instances) and I worried I'd just be reimplementing a lot of stuff that the validate functionality will handle.

I think the choices we have are:

1. Implement our own mini validation in a short-term convenient way, as a last ditch effort to prevent crashes/not useful error messages.
2. Pause this PR until validation is complete.
3. Go ahead and just implement the final solution now and not worry that the error message is unhelpful knowing that eventually validation will come and make it better.

I was worried about the Monday deadline for the beta release so I didn't want to do 2 or 3. Hence, I went with 1.

Reply:
Thanks, makes sense. We do actually need a similar check which is currently missing in here, when an import targeting a specific index does not correspond with the indexes in the config, the import will fail silently. We can address that in a separate PR though to move this one along.